test/videojs-contrib-hls.test.js

```diff
@@ -1135,6 +1143,7 @@ QUnit.test('playlist 404 should blacklist media', function(assert) {
   assert.equal(this.env.log.warn.args[1],
               'Problem encountered with the current HLS playlist. Trying again since it is the final playlist.',
               'log specific error message for final playlist');
+  assert.equal(retryplaylist, 1, 'retried final playlist for once');
```

Comment:
I think you should add a check here to make sure `blacklistplaylist` is still `1`